machine/v1beta1/types_powervsprovider.go

```diff
@@ -0,0 +1,156 @@
+
+	// Zone indicates the Power VS ServiceInstance zone.
+	// +optional
+	Zone string `json:"zone,omitempty"`
+
+	// ServiceInstance is the reference to the Power VS ServiceInstance on which the machine instance will be created.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// Image is the OS image to use to create the instance.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// Network is the reference to the Network to use for this instance.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// KeyPairName is the name of the KeyPair to use for SSH
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
+
+	// SysType is the System type used to host the instance.
+	// Defaults to s922.
+	// +optional
+	SysType string `json:"sysType,omitempty"`
+
+	// ProcessorType is the VM instance processor type.
+	// It must be set to one of the following values: dedicated, shared or capped.
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:=dedicated;shared;capped
+	// +optional
+	ProcType string `json:"procType,omitempty"`
+
+	// Processors is the number of virtual processors in a virtual machine.
+	// Defaults to 0.5.
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// Memory is the size of a virtual machine's memory, in GB.
+	// Defaults to 32 GB.
+	// +optional
+	Memory int32 `json:"memory,omitempty"`
+}
+
+// PowerVSResource is a reference to a specific PowerVS resource by ID, Name or RegEx
+// Only one of ID, Name or RegEx may be specified. Specifying more than one will result in
+// a validation error.
+type PowerVSResource struct {
+	// ID of resource
+	// +optional
+	ID *string `json:"id,omitempty"`
+
+	// Name of resource
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// Regex to find resource
+	// +optional
+	RegEx *string `json:"regex,omitempty"`
+}
+
+// PowerVSMachineProviderConfigList contains a list of PowerVSMachineProviderConfig
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type PowerVSMachineProviderConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PowerVSMachineProviderConfig `json:"items"`
+}
+
+// PowerVSMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains PowerVS-specific status information.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type PowerVSMachineProviderStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// InstanceID is the instance ID of the machine created in PowerVS
+	// +optional
+	InstanceID *string `json:"instanceId,omitempty"`
```

Comment:
When will this be set, what does it mean if it is missing

Reply:
InstanceID uniquely identifies a VM instance under a Power VS Service, Once the VM is created in Power VS Service the InstanceID is set and later it is used to update the VM or delete the VM.

If InstanceID is missing we need to make a call to PowerVS cloud with VM name to identify a instance, In our case its more time consuming than querying with ID.